client/state/domains/management/reducer.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import { merge } from 'lodash';
+/**
+ * Internal dependencies
+ */
+import { createReducer, combineReducers } from 'state/utils';
+import { domainWhoisSchema } from './schema';
+import {
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_RECEIVE,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_FAILURE,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_SUCCESS,
+	DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_UPDATE,
+	DOMAIN_MANAGEMENT_WHOIS_RECEIVE,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST_FAILURE,
+	DOMAIN_MANAGEMENT_WHOIS_REQUEST_SUCCESS,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE_FAILURE,
+	DOMAIN_MANAGEMENT_WHOIS_SAVE_SUCCESS,
+	DOMAIN_MANAGEMENT_WHOIS_UPDATE
+} from 'state/action-types';
+
+/**
+ * Returns the updated requests state after an action has been dispatched. The
+ * state maps domain to whether a request is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const requesting = createReducer( false, {
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST ]: ( state ) => ( { ...state, _contactDetailsCache: true } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_SUCCESS ]: ( state ) => ( { ...state, _contactDetailsCache: false } ),
+	[ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_FAILURE ]: ( state ) => ( { ...state, _contactDetailsCache: false } ),
+	[ DOMAIN_MANAGEMENT_WHOIS_REQUEST ]: ( state, { domain } ) => ( { ...state, [ domain ]: true } ),
+	[ DOMAIN_MANAGEMENT_WHOIS_REQUEST_SUCCESS ]: ( state, { domain } ) => ( { ...state, [ domain ]: false } ),
+	[ DOMAIN_MANAGEMENT_WHOIS_REQUEST_FAILURE ]: ( state, { domain } ) => ( { ...state, [ domain ]: false } )
+} );
```

Comment:
This reducer seems to be caring about two separate concerns:

1. Holding (presumably domain-agnostic) cache request state:

```js
import { stubFalse, stubTrue } from 'lodash';

const cache = createReducer( false, {
  [ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST ]: stubTrue,
  [ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_SUCCESS ]: stubFalse,
  [ DOMAIN_MANAGEMENT_CONTACT_DETAILS_CACHE_REQUEST_FAILURE ]: stubFalse,
} );
```

2. Holding domain-specific whois request state. A good candidate for `keyedReducer`:

```js
const whois = keyedReducer( 'domain', createReducer( false, {
  [ DOMAIN_MANAGEMENT_WHOIS_REQUEST ]: stubTrue,
  [ DOMAIN_MANAGEMENT_WHOIS_REQUEST_SUCCESS ]: stubFalse,
  [ DOMAIN_MANAGEMENT_WHOIS_REQUEST_FAILURE ]: stubFalse,
} ) );

3. Which can be combined:

```js
// also note the preference for `isRequesting` over `requesting`
const isRequesting = combineReducers( { cache, whois } );
```

Analogous separation of reducer concern and analogous usage of `keyedReducer` are recommended for the following domain-keyed reducers (`saving`, `items`).